src/main/java/org/codehaus/mojo/exec/LineRedirectOutputStream.java

```diff
@@ -32,11 +35,17 @@
  */
 class LineRedirectOutputStream extends OutputStream {
 
-    private StringBuilder currentLine = new StringBuilder();
     private final Consumer<String> linePrinter;
+    private final Charset charset;
+    private ByteArrayOutputStream buffer = new ByteArrayOutputStream();
 
     public LineRedirectOutputStream(Consumer<String> linePrinter) {
-        this.linePrinter = linePrinter;
+        this(linePrinter, Charset.defaultCharset());
+    }
+
+    public LineRedirectOutputStream(Consumer<String> linePrinter, Charset charset) {
```

Comment:
I figured this could be useful to have public even though it is, for now, only used in tests. Let me know if you want me to put it package protected.

Reply:
can be public